packages/components/stencil-angularjs-output-target/src/output-angularjs.ts

```diff
@@ -0,0 +1,397 @@
+} from './constants';
+import { OutputTargetAngularJS } from './types';
+
+/**
+ * Get components apart from excluded ones.
+ * @param  excludeComponents Components to exclude
+ * @param  components        Source library components
+ * @return                   Filtered components to work on
+ */
+function getFilteredComponents(
+  excludeComponents: string[] = [],
+  components: ComponentCompilerMeta[],
+) {
+  return components
+    .sort((cmpA, cmpB) => {
+      if (cmpA.tagName < cmpB.tagName) {
+        return -1;
+      }
+
+      return cmpA.tagName > cmpB.tagName ? 1 : 0;
+    })
+    .filter((c) => !excludeComponents.includes(c.tagName) && !c.internal);
+}
+
+/**
+ * Function set loader function imports and calls, based on output target configuration
+ * @param  config       Stencil source library configuration
+ * @param  outputTarget Output target configuration
+ * @return              Text to write into final proxy file which includes code to call loader methods
+ */
+function getLoaderImports(
+  config: Config,
+  outputTarget: OutputTargetAngularJS,
+): string {
+  if (!outputTarget.applyPolyfills && !outputTarget.defineCustomElements) {
+    return '';
+  }
+
+  const loaderImportPath = getPathToCorePackageLoader(config, outputTarget);
+  const imports = `import { ${
+    outputTarget.applyPolyfills ? `${APPLY_POLYFILLS}, ` : ''
+  }${DEFINE_CUSTOM_ELEMENTS} } from '${loaderImportPath}';\n`;
+  const calls = `${
+    outputTarget.applyPolyfills
+      ? `${APPLY_POLYFILLS}().then(() => { ${DEFINE_CUSTOM_ELEMENTS}(); });`
+      : `${DEFINE_CUSTOM_ELEMENTS}();\n`
+  }`;
+
+  return `${imports}${calls}`;
+}
+
+/**
+ * Get the component model, based on mutable properties and watchers callback
+ * @param  component The component to search in
+ * @return           Object describing component model (event and prop attached)
+ */
+function getComponentModelInformation(component: ComponentCompilerMeta) {
+  const { events, properties, watchers } = component;
+
+  const mutableProperties = properties.filter((prop) => prop.mutable);
+
+  const modelProperty = mutableProperties.find((prop) =>
+    watchers.find((watcher) => watcher.propName === prop.name),
+  );
+
+  if (!modelProperty) {
+    return {};
+  }
+
+  const modelEvent = events.find((event) =>
+    event.name.startsWith(modelProperty.name),
+  );
+
+  if (!modelEvent) {
+    return {};
+  }
+
+  return {
+    propName: modelProperty.name,
+    eventName: modelEvent.name,
+  };
+}
+
+/**
+ * Get the component properties for wrapper component bindings
+ * @param  properties Component properties
+ * @return            AngularJS bindings to write
+ */
+function getComponentProperties(properties: any[]): string {
+  return properties
+    .sort((propA, propB) => {
+      if (propA.name < propB.name) {
+        return -1;
+      }
+
+      return propA.name > propB.name ? 1 : 0;
+    })
+    .map((prop) =>
+      prop.isModel
+        ? 'model: "<", \n\t\tonModelChange: "&"'
```

Comment:
Will this work with all OSes ?

Reply:
Hmm yes good point, it seems that it will work for all OSes, unless you're on old MacOS (9 and older ones)

Reply:
All `\n` or `\t` have been replaced by direct return or simple whitespaces.